src/Promitor.Core/Metrics/MeasuredMetric.cs

```diff
@@ -63,9 +63,11 @@ public static MeasuredMetric CreateForDimension(double? value, string dimensionN
         {
             Guard.NotNullOrWhitespace(dimensionName, nameof(dimensionName));
             Guard.NotNull(timeseries, nameof(timeseries));
-            Guard.For<ArgumentException>(() => timeseries.Metadatavalues.Any() == false);
+            
+            var dimensionMetadataValue = timeseries.Metadatavalues.Where(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);            
+            Guard.For<ArgumentException>(() => dimensionMetadataValue.Count() == 0);
 
-            var dimensionValue = timeseries.Metadatavalues.Single(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);
+            var dimensionValue = dimensionMetadataValue.First();
```

Comment:
Can you elaborate on the reasoning of this change?

Reply:
Are you referring the the reason for this PR overall or the reason why I wrote these lines of code in this way? 

As for reasoning for this PR - it's basically what I wrote in the PR description. Azure Monitor would return time series without the requested dimension value throughout the day, and the current version of Promitor does not handle that well; whenever that happens, we see a complete metric gap for 5-10 mins. The thought is then we can catch whenever that happens and just skip over the offending time series. This would fix our metric gap for the most part. 

As for why I wrote the code this way - I'm basically checking for the a dimension value match first; if no match I then throw an exception and catch it. This is my first time writing C# though, so please advise if I'm doing it right.

Reply:
As an alternative to just skipping over the offending time series, I'm wondering if it makes sense to write our the time series without the requested dimension value. I'm not too sure there, because I don't know if returning time series without a dimension value expected behavior or a bug.

Reply:
> As for why I wrote the code this way - I'm basically checking for the a dimension value match first; if no match I then throw an exception and catch it. This is my first time writing C# though, so please advise if I'm doing it right.

This is what I was looking for, thanks.

Can you add some unit tests in a new file in https://github.com/tomkerkhove/promitor/tree/master/src/Promitor.Tests.Unit/Metrics please?